enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/discovery/SrvRecordResolver.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.causalclustering.discovery;
+
+import javax.naming.NamingException;
+
+import org.neo4j.helpers.AdvertisedSocketAddress;
+
+public interface SrvRecordResolver
```

Comment:
This will allow me to write a unit test (by providing a `MockSrvRecordResolver`).
As I look at writing the unites I realise that I should move logic out of the `SrvRecordResolver` and into the `SrvHostnameResolver` so that as much of it as possible is unit tested